joinmarket-webui/jam
# refactor: prepare spending fidelity bond

This is a refactoring in preperation of #547 

Why? Because the follow-up PR would contain too much changes to be properly reviewed.

The change set may seem a little bit overwhelming, but this PR is in fact just renaming and moving stuff around.

e.g.
- more generic components  (especially regarding passing translations strings) that will be used in PR for #547
- add missing i18n translation keys
- add property`utxosByJar` to `WalletInfo` (because it is used quite frequently and should only be created once)
- renaming `account` -> `jar` wherever possible

Also, it adds a `globals.d.ts` file to contain all app-wide used TypeScript types (currently only holding `JarIndex`).

One small bugfix is included: Prevent an error while creating a Fidelity Bond from displaying a loading indicator indefinitely.
Before this fix, when creating a FB failed, a user could not restart the procedure. After this PR is applied, a user can properly restart the flow in order to try creating a FB again.

### How to test?
Try creating a fidelity bond with a coinbase transaction that has less than 100 confirmations (execute `./docker/regtest/mine-block 1 ${address_from_jar_A}`) without this fix applied (e.g. from the current master branch) and notice that the resulting error prevents users from restarting the "create FB"-flow. Then apply this fix (by checking out this branch `git checkout prepare-spend-fb`) and try again. This time the error should be displayed and a button to "Start over" is presented.
